spec/prometheus_spec.lua

```diff
@@ -0,0 +1,41 @@
+
+describe('prometheus', function()
+  before_each(function() package.loaded['apicast.prometheus'] = nil end)
+
+  describe('shared dictionary is missing', function()
+    before_each(function() ngx.shared.prometheus_metrics = nil end)
+
+    it('can be called', function()
+      assert.is_nil(require('apicast.prometheus')())
+    end)
+
+    it('can be collected', function()
+      assert.is_nil(require('apicast.prometheus'):collect())
+    end)
+  end)
+
+  describe('shared dictionary is there', function()
+    before_each(function()
+      ngx.shared.prometheus_metrics = {
+        set = function() end,
+        get_keys = function() return {} end
+      }
+    end)
+
+    local prometheus
+
+    before_each(function()
+      prometheus = assert(require('apicast.prometheus'))
+    end)
+
+    it('can be called', function()
```

Comment:
Should we change this to test that it delegates `counter`, `gauge`, etc. to the Prometheus lib?

Reply:
Yep, makes sense.

Reply:
Done.